utils/plugin/plugin_downloader_test.go

```diff
@@ -0,0 +1,225 @@
+package plugin
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/argoproj/argo-rollouts/utils/config"
+
+	"github.com/argoproj/argo-rollouts/utils/defaults"
+	"github.com/tj/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+type MockFileDownloader struct {
+	FileDownloader
+}
+
+func (m MockFileDownloader) Get(url string) (*http.Response, error) {
+	responseBody := io.NopCloser(bytes.NewReader([]byte(`test`)))
+	return &http.Response{
+		Status:        "200",
+		StatusCode:    200,
+		Proto:         "HTTP/1.1",
+		ProtoMajor:    1,
+		ProtoMinor:    1,
+		Header:        nil,
+		Body:          responseBody,
+		ContentLength: 4,
+	}, nil
+}
+
+func TestNotInitialized(t *testing.T) {
+	_, err := config.GetConfig()
+	assert.Error(t, err)
+}
+
+func TestInitPlugin(t *testing.T) {
```

Comment:
Small suggestion but personally I prefer BDD style test for readability. [Example](https://github.com/argoproj/argo-cd/blob/master/server/extension/extension_test.go#L164)

Reply:
Will move to this on the traffic router plugin branch